### PR TITLE
Several STOPS-related fixes

### PR DESCRIPTION
**Pull request by EliAndrewC**
I don't believe that any of this conflicts with any of the other PRs, so the order in which things are merged shouldn't matter.

I'll explain my fixes inline.

### CODE REVIEW

File: uber/models.py [hunk 1]
```diff
@@ -1062,7 +1062,7 @@ def _misc_adjustments(self):
         if c.AT_THE_CON and self.badge_num and self.is_new:
             self.checked_in = datetime.now(UTC)
```
Comment (EliAndrewC) [inline uber/models.py]: Until now, you couldn't ever update someone's age group once it was set!  On my dev server I tried changing someone to no longer be under 18, but it just kept telling me they couldn't be a volunteer no matter what I changed their birthdate to.

I can't think of a time where the Attendee has a birthdate set and has passed all validations (so we know the birthdate is valid) but we wouldn't want to update the ``age_group``.  I've therefore changed this to just always set it.

File: uber/models.py [hunk 2]
```diff
@@ -1144,16 +1144,14 @@ def badge_cost(self):
 
     @property
     def age_group_conf(self):
```
Comment (EliAndrewC) [inline uber/models.py]: We perform validations against this property, but we don't update ``age_group`` until we're already passed validations.  Therefore, when both the ``age_group`` and the ``birthdate`` are set, we want to use the ``birthdate`` to determine which of the ``AGE_GROUP_CONFIGS`` we select.  Otherwise we'd run into a situation such as the following
- attendee exists who is wrongly marked as under 18 and who is not signed up for volunteering
- admin corrects their birthdate and marks them as volunteering
- validation checks ``age_group`` which has not yet updated, and says they cannot volunteer

This fixes that problem.

File: uber/site_sections/jobs.py [hunk 3]
```diff
@@ -168,6 +168,7 @@ def staffers_by_job(self, session, id, message=''):
             'message':   message,
             'attendees': session.query(Attendee.id, Attendee.full_name)
                                 .filter_by(staffing=True, **({'trusted': True} if job.restricted else {}))
```
Comment (EliAndrewC) [inline uber/site_sections/jobs.py]: This is the fix for #1264

File: uber/templates/regform.html [hunk 4]
```diff
@@ -161,17 +161,17 @@
         </div>
         {% popup_link "../static_views/affiliates.html" "What's an affiliate?" %}
     </div>
-{% endif %}
```
Comment (EliAndrewC) [inline uber/templates/regform.html]: I originally left the shirt row outside of the if-block because we wanted it on the admin form.  However, we already had it separately on the admin form, which meant it was showing up twice.  This was causing problems on the admin form, since the shirt would be submitted twice, and the backend would complain if the size changed, since it would only be changed in one of the two places, and it would be parsed as a list rather than an integer, etc.

This change just moves it inside the if block so on the admin form it doesn't get generated here.

File: uber/templates/registration/form.html [hunk 5]
```diff
@@ -84,12 +84,18 @@
         }
     };
 
-    $(document).ready(function(){
```
Comment (EliAndrewC) [inline uber/templates/registration/form.html]: We want "Trusted" and "Assigned Depts" to only show up if the volunteering checkbox is checked.  That wasn't implemented before (it's how it worked last year, before the refactor, so this turns it back on).

File: uber/templates/registration/form.html [hunk 6]
```diff
@@ -199,7 +205,15 @@
 
 {% include "regform.html" %}
 
-<div class="form-group">
```
Comment (EliAndrewC) [inline uber/templates/registration/form.html]: This is the fix for #1263

File: uber/templates/registration/form.html [hunk 7]
```diff
@@ -270,7 +284,6 @@
         <label for="shirt" class="col-sm-2 control-label">Shirt Size</label>
         <div class="col-sm-6">
             <select name="shirt" class="form-control">
```
Comment (EliAndrewC) [inline uber/templates/registration/form.html]: We don't need this option on the admin form, since ``c.SHIRT_OPTS`` already includes the ``0`` option.

File: uber/models.py [hunk 8]
```diff
@@ -1144,16 +1144,14 @@ def badge_cost(self):
...
-        if self.age_group and self.age_group != c.AGE_UNKNOWN:
-            return c.AGE_GROUP_CONFIGS[self.age_group]
-        elif self.birthdate:
+        if self.birthdate:
             day = c.EPOCH.date() if date.today() <= c.EPOCH.date() else sa.localized_now().date()
             attendee_age = (day - self.birthdate).days / 365.2425
```
Comment (binary1230) [inline uber/models.py]: this /365 here should be converted to some kind of native date handling thing to deal with leap years and junk more precisely.  discussion in https://github.com/magfest/ubersystem/issues/1125, I'll add a note there.

don't hold up this PR for it though, it's just something I noticed

Comment (binary1230) [inline uber/models.py]: oh wait duh this is the exact same line as the bug is referencing, so, ignore, we already know about it.

**Comment by binary1230** [comment]
+1

File: uber/templates/regform.html (hunk 4 above)
Comment (kitsuta) [inline uber/templates/regform.html]: Oh no, this was my fault - I added the shirt size option in another PR because I thought it wasn't being displayed on the admin form.

Before we merge this, I'd like to undo the change you made here and remove my shirt row on the admin form - this is the better solution.

Comment (kitsuta) [inline uber/templates/regform.html]: Oops, this was my fault - I didn't realize this was being displayed on the admin form.
